kornia/image/image.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+from torch.utils.dlpack import from_dlpack, to_dlpack
+
+from kornia.core import Device, Dtype, Tensor
+from kornia.image.base import ChannelsOrder, ImageLayout, ImageSize, PixelFormat
+from kornia.io.io import ImageLoadType, load_image, write_image
+
+# TODO: move this utils to kornia.image.conversions
+from kornia.utils.image import image_to_tensor, tensor_to_image
+
+# placeholder for numpy
+np_ndarray = Any
+DLPack = Any
+
+
+class Image:
+    r"""Class that holds an Image Tensor representation.
+
+    .. note::
+
+        Disclaimer: This class provides the minimum functionality for image manipulation. However, as soon
+        as you start to experiment with advanced tensor manipulation, you might expect fancy
+        polymorphic behaviours.
+
+    .. warning::
+
+        This API is experimental and might suffer changes in the future.
+
+    Args:
+        data: a torch tensor containing the image data.
+        layout: a dataclass containing the image layout information.
+
+    Examples:
+        >>> # from a torch.tensor
+        >>> data = torch.randint(0, 255, (3, 4, 5))  # CxHxW
+        >>> layout = ImageLayout(
+        ...     image_size=ImageSize(4, 5),
+        ...     channels=3,
+        ...     pixel_format=PixelFormat.RGB,
+        ...     channels_order=ChannelsOrder.CHANNELS_FIRST,
+        ... )
+        >>> img = Image(data, layout)
+        >>> assert img.channels == 3
+
+        >>> # from a numpy array (like opencv)
+        >>> data = np.ones((4, 5, 3), dtype=np.uint8)  # HxWxC
+        >>> img = Image.from_numpy(data, pixel_format=PixelFormat.BGR)
+        >>> assert img.channels == 3
+        >>> assert img.width == 5
+        >>> assert img.height == 4
+    """
+
+    def __init__(self, data: Tensor, layout: ImageLayout) -> None:
+        """Image constructor.
+
+        Args:
+            data: a torch tensor containing the image data.
+            layout: a dataclass containing the image layout information.
+        """
+        self._data = data
+        self._layout = layout
+
+    def __repr__(self) -> str:
+        return f"Image data: {self.data}\nLayout: {self.layout}"
```

Comment:
what do you think about having some fields for pixel value stats like value range, mean etc? ( i can add if you think its a good idea)

Reply:
sounds good!